cmd/pulumictl/create/dev-version/dev-version.go

```diff
@@ -0,0 +1,96 @@
+package devversion
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/v32/github"
+	gh "github.com/pulumi/pulumictl/pkg/github"
+	"github.com/pulumi/pulumictl/pkg/util"
+	"github.com/spf13/cobra"
+	viperlib "github.com/spf13/viper"
+)
+
+var (
+	org         string
+	repo        string
+	tokenClient *http.Client
+)
+
+type Payload struct {
+	DevVersion string `json:"dev_version"`
+}
+
+func Command() *cobra.Command {
+	viper := viperlib.New()
+	command := &cobra.Command{
+		Use:   "dev-version [version]",
+		Short: "Create a dev version",
+		Long:  `Send a repository dispatch payload to the docs repo`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Grab all the configuration variables
+			githubToken := viperlib.GetString("token")
+			org = viper.GetString("org")
+			docsRepo := viper.GetString("docs-repo")
+			eventType := viper.GetString("event-type")
+			version := args[0]
+
+			// perform some string manipulation and validation
+			docsRepoArray := strings.Split(docsRepo, "/")
+
+			// if the string split doesn't return 2 values, it's probably not right
+			if len(docsRepoArray) != 2 {
+				return fmt.Errorf("unable to use docs repo:"+
+					" format must be <org>/<repo> - value: %s",
+					docsRepo)
+			}
+
+			// create a github client and token
+			ctx, client := gh.CreateGithubClient(githubToken)
+
+			// create the JSON payload
+			jsonPayload, err := json.Marshal(Payload{
```

Comment:
It's a little sad how this is basically the same as `dispatch` except for the payload. 
I wonder if we should instead just extend dispatch to be more general and take a ref or a key=value list?

Like `dispatch str` would assume str is a ref version, but `dispatch k=v` would see the `=` and instead just treat it as a key value, allowing multiple if wanted `dispatch k1=v1 k2=v2`.

That would cover this case and I suspect would cover any other dispatch needs we have in the future as well without breaking the current uses (because no semver has an `=` in it)

Reply:
Yeah that makes more sense, I'll change that.